Sources/SwiftBSON/BSON.swift

```diff
@@ -95,6 +95,26 @@ public enum BSON {
             self = int64.bson
             return
         }
+
+        if let uuidJSON = try json.unwrapObject(withKey: "$uuid", keyPath: keyPath) {
+            guard let uuidString = uuidJSON.stringValue else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Could not parse `uuid` from \"\(uuidJSON)\", "
```

Comment:
Can we make this description a little more specific and indicate that we were expecting a string but got some other value.